### PR TITLE
feat(): Add demo to config template & avoid some code if demo mode is true

### PR DESCRIPTION
**Pull request by qperie**
## Describe your changes

- Adds "demo" property to conf.json.template (was already used as conf but not present in template).
- Skips creation of httpClients for each domain when demo mode is activated. Indeed, in demo mode we display "fake" cards based on raw data directly written in a java file. We do not need to get data from the differents domains (GAR, PMB, ...)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by aymericdevriesere — APPROVED** [review]

File: src/main/java/fr/openent/gar/Gar.java [hunk 1]
```diff
@@ -26,6 +26,9 @@ public void start() throws Exception {
 		super.start();
 		final EventBus eb = getEventBus(vertx);
 
+		demo = config.getBoolean("demo", false);
+		CONFIG = config;
```
Comment (BastienBenard) [inline src/main/java/fr/openent/gar/Gar.java]: Il me semble que CONFIG ne respecte pas les convention de nommage, c'est en uppercase quand c'est en final static. Est ce que tu peux faire un refacto ou ça casse tout ?

File: src/main/java/fr/openent/gar/Gar.java [hunk 2]
```diff
@@ -44,7 +47,8 @@ public void start() throws Exception {
 		}
 
 		final JsonObject garRessources = config.getJsonObject("gar-ressources", new JsonObject());
-		if (garRessources.containsKey("cert") || garRessources.containsKey("key")) {
+		if ((garRessources.containsKey("cert") && !garRessources.getString("cert").isEmpty()) ||
+				(garRessources.containsKey("key") && !garRessources.getString("key").isEmpty())) {
```
Comment (BastienBenard) [inline src/main/java/fr/openent/gar/Gar.java:50]: tu peux utilisé 
`!Strings.isNullOrEmpty(garRessources.getString("cert")) || !Strings.isNullOrEmpty(garRessources.getString("key"))`

File: src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java [hunk 3]
```diff
@@ -53,13 +53,15 @@ public DefaultResourceService(Vertx vertx, JsonObject garRessource, JsonObject i
...
+        if (!Gar.demo) {
+            for (String domain : domains.fieldNames()) {
+                final JsonObject res = domains.getJsonObject(domain);
+                if (res == null) continue;
+                try {
+                    httpClientByDomain.put(domain, generateHttpClient(new URI(garHost), res.getString("cert"), res.getString("key")));
+                } catch (URISyntaxException e) {
+                    log.error("[DefaultResourceService@constructor] An error occurred when creating the URI", e);
+                }
             }
```
Comment (BastienBenard) [inline src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java]: tu peux peut etre utilisé un Stream ici, avec un filter pour supprimer la ligne 59

**Review by BastienBenard — COMMENTED** [review]
Il y a tout plein de magic String

File: src/main/java/fr/openent/gar/Gar.java (hunk 2 above)
Comment (BastienBenard) [reply src/main/java/fr/openent/gar/Gar.java:50]: My bad ce n'est pas utilisable

File: src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java [hunk 4]
```diff
@@ -53,13 +53,15 @@ public DefaultResourceService(Vertx vertx, JsonObject garRessource, JsonObject i
...
-            } catch (URISyntaxException e) {
-                log.error("[DefaultResourceService@constructor] An error occurred when creating the URI", e);
+        if (!Gar.demo) {
+            for (String domain : domains.fieldNames()) {
+                final JsonObject res = domains.getJsonObject(domain);
+                if (res == null) continue;
+                try {
+                    httpClientByDomain.put(domain, generateHttpClient(new URI(garHost), res.getString("cert"), res.getString("key")));
+                } catch (URISyntaxException e) {
+                    log.error("[DefaultResourceService@constructor] An error occurred when creating the URI", e);
```
Comment (qpic-cgi) [inline src/main/java/fr/openent/gar/service/impl/DefaultResourceService.java]: concatène ton erreur plutôt (String.fortmat("%s %s", "le message", e);

**Pull request merged**